pkg/volume/csi/csi_attacher.go

```diff
@@ -220,23 +218,17 @@ func (c *csiAttacher) VolumesAreAttached(specs []*volume.Spec, nodeName types.No
 			klog.Error(log("attacher.VolumesAreAttached missing volume.Spec"))
 			return nil, errors.New("missing spec")
 		}
-		source, err := getCSISourceFromSpec(spec)
+		driverName, volumeHandle, _, _, err := getCSIFieldsFromSpec(spec)
 		if err != nil {
 			klog.Error(log("attacher.VolumesAreAttached failed: %v", err))
 			continue
 		}
-		skip, err := c.plugin.skipAttach(source.Driver)
```

Comment:
We should keep the skipAttach checks?

Reply:
the Plugin.CanAttach(spec) is called by the controller to determine if a plugin supports attachment.  When false, none of this happens. Keeping this maybe redundant calls.

Reply:
Putting skipping back.